first_order_model/shrink_util.py

```diff
@@ -1060,43 +1109,49 @@ def compute_deletion(layer_graph,
         
-        # Counts up for each index in the inputs for this layer
-        counter = 0
+        # base counts up for each index in the inputs for this layer
+        # This is to solve the following problem:
+        # If layer a, and b get concatted into the inputs of c, then we want to note that
+        # deleting from b should only affect its corresponding inputs in c.
+        # We use this index to track the "base" of an input. So if layer a has length 20, layer b has length 10
+        # Then when looking at layer b in the following loop, base=20, so you know it affects indices
+        # after 20.
+        base = 0
         for previous_layer in layer_graph[following_layer].before:
 
-            # If you don't delete from the previous layer, just increment the counter by its ouputs.
+            # If you don't delete from the previous layer, just increment the base by its ouputs.
             if previous_layer not in deletions:
-                counter += layer_graph[previous_layer].out_channels
+                base += layer_graph[previous_layer].out_channels
                 continue
 
-            # If delete from previous layer, delete corresponding element from this layer and increment the counter
+            # If delete from previous layer, delete corresponding element from this layer and increment the base
             for deletion in deletions[previous_layer]:
                 deletions[following_layer].append(
-                    (counter + deletion[0], counter + deletion[1]))
-            counter += layer_graph[previous_layer].out_channels
+                    (base + deletion[0], base + deletion[1]))
+            base += layer_graph[previous_layer].out_channels
 
         # If there is another tied after layer i.e. a layer who has their output tied to the input of this layer, 
         # like in resnet, we need to delete from their outputs as well
         if len(layer_graph[following_layer].tied_after) != 0:
-            total_deleted = copy.copy(deletions[following_layer])
+            filters_removed_from_following_layer = copy.copy(deletions[following_layer])
 
-            # Add the output layer to our work queue (custom deletions)
-            for after_tied in layer_graph[following_layer].tied_after:
-                if after_tied not in deleted_things:
-                    deleted_things.add(after_tied)
-                    custom_deletions.append((after_tied, total_deleted, 'tie'))
+            # Add the output layer to our work queue
+            for dependent_layer in layer_graph[following_layer].tied_after:
+                if dependent_layer not in deleted_things:
+                    deleted_things.add(dependent_layer)
+                    filters_to_recursively_remove.append((dependent_layer, filters_removed_from_following_layer, 'tie'))
 
     # The original layer is a special case because you delete its output not input
     deletions[layer].insert(0, 'first')
```

Comment:
I don't understand what this is doing. Everytime you add the very first layer when computing deletions? can you clarify?

Reply:
I added some more comments and pointed to where this is used.